packages/react/src/componentsV2/baseCheckout/actions.ts

```diff
@@ -0,0 +1,112 @@
+import { Currency, CustomerDetails, ProductDetails } from '@heliofi/common';
+import { HelioSDK as HelioSDKType } from '@heliofi/sdk';
+import { removeUndefinedFields } from '../../utils';
+import { InheritedOnSumbit } from './constants';
+import { Option } from '../../components/selectbox';
+
+export const getInitialValues = (
+  paymentDetails: any,
+  normalizedPrice: number,
+  canSelectCurrency: boolean,
+  initialCurrency?: string
+) => ({
+  requireEmail: paymentDetails?.features.requireEmail,
+  requireDiscordUsername: paymentDetails?.features.requireDiscordUsername,
+  requireFullName: paymentDetails?.features.requireFullName,
+  requireTwitterUsername: paymentDetails?.features.requireTwitterUsername,
+  requirePhoneNumber: paymentDetails?.features.requirePhoneNumber,
+  requireCountry: paymentDetails?.features.requireCountry,
+  requireDeliveryAddress: paymentDetails?.features.requireDeliveryAddress,
+  requireProductDetails: paymentDetails?.features.requireProductDetails,
+  canChangePrice: paymentDetails?.features.canChangePrice,
+  canChangeQuantity: paymentDetails?.features.canChangeQuantity,
+  fullName: undefined,
+  email: undefined,
+  discordUsername: undefined,
+  twitterUsername: undefined,
+  country: undefined,
+  areaCode: '',
+  deliveryAddress: undefined,
+  city: undefined,
+  street: undefined,
+  streetNumber: undefined,
+  phoneNumber: undefined,
+  quantity: paymentDetails?.features.canChangeQuantity ? 1 : undefined,
+  customPrice: paymentDetails?.features.canChangePrice
+    ? undefined
+    : normalizedPrice,
+  canSelectCurrency,
+  currency: canSelectCurrency ? undefined : initialCurrency,
+  productValue: undefined,
+});
+
+export const getCurrency = (currencyList: any[], currency?: string) => {
+  if (!currency) return null;
+  return currencyList.find((c: any) => c.symbol === currency);
+};
+
+export const handleSubmit =
+  ({
+    paymentDetails,
+    HelioSDK,
+    price,
+    onSubmit,
+    currencyList,
+  }: {
+    paymentDetails?: any;
+    HelioSDK: HelioSDKType;
+    price: number;
+    onSubmit: InheritedOnSumbit;
+    currencyList: Currency[];
+  }) =>
+  (values: any) => {
```

Comment:
is it really any ?

Reply:
I think there is type FormikValues if this is coming from a formik form